### PR TITLE
vkconfig: fix clang build

### PR DESCRIPTION
**Pull request by christophe-lunarg**
It looks like I messed up!

### CODE REVIEW

File: vkconfig_core/util.h [hunk 1]
```diff
@@ -55,7 +55,7 @@ bool IsNumber(const std::string& s);
 
 template <typename T>
 typename std::vector<T>::iterator FindItByKey(std::vector<T>& container, const char* key) {
-    assert(key != nullptr && key != "");
+    assert(key != nullptr);
```
Comment (lunarpapillo) [inline vkconfig_core/util.h:58]: Is the change in behavior intentional?  The original assertion could be rewritten as something like:
```
assert(key != nullptr && key[0] != '\0');
```
or:
```
assert(key != nullptr && std::strcmp(key, "") != 0);
```
right?

**Review by lunarpapillo — COMMENTED** [review]
The switch of `strcmp` to `std::strcmp` looks good.  The other changes alter behavior; if that's intentional, it's fine with me.

File: vkconfig_core/util.h (hunk 1 above)
Comment (christophe-lunarg) [reply vkconfig_core/util.h:58]: Yes expect, key != "" cause a warning as this is undefined

Comment (lunarpapillo) [reply vkconfig_core/util.h:58]: I do see why `key != ""` would not work.  But removing the check entirely changes behavior, which I don't expect from a change intended only to fix compiler errors, as there were other ways to change the code that would have preserved the original behavior.

If the behavior change is intentional, I'm okay with it.

Comment (lunarpapillo) [reply vkconfig_core/util.h:58]: Why was the check for an empty string removed entirely, instead of being fixed to use `std::strcmp`?

Comment (christophe-lunarg) [reply vkconfig_core/util.h:58]: Ah good point, my bad. I've fixed it with upcoming PR.

**Pull request merged**